utils/nba_api_client.py

```
@@ -0,0 +1,175 @@
+        if isinstance(payload, dict):
+            return {k: self._deserialize(v) for k, v in payload.items()}
+        if isinstance(payload, list):
+            return [self._deserialize(v) for v in payload]
+        return payload
+
+    def _cache_path(self, name, params):
+        key_str = json.dumps({"name": name, "params": params}, sort_keys=True, default=str)
+        hashed = hashlib.sha256(key_str.encode("utf-8")).hexdigest()
+        return os.path.join(self.cache_dir, f"{name}_{hashed}.json")
+
+    def _read_cache(self, name, params):
```

Ruff (0.14.10)

Missing return type annotation for private function `_read_cache`

(ANN202)